ote/src/cljs/ote/app/controller/transit_visualization.cljs

```diff
@@ -187,6 +175,13 @@
                 :gtfs-package-info (:gtfs-package-info response)
                 :route-hash-id-type (:route-hash-id-type response)))
 
+(defn- init-view-state [app]
+  (let [initial-view-state {:all-route-changes-checkbox nil
+                            :all-route-changes-display? false
+                            :open-sections {:gtfs-package-info false}
+                            :service-changes-for-date-loading? true}]
+    (assoc app :transit-visualization initial-view-state)))
+
```

Comment:
This is good. But you still could do one more thing.

In transit-changes.clj controller at row 30 forwards you could add :transit-visualization clean up.
Like so:
Replace {:path [:transit-changes]} -> {}

And from row 34
 (-> app
      (assoc-in [:transit-changes :loading?] true)
      (assoc-in [:transit-changes :show-all] true)
      (dissoc :transit-visualization))

Reply:
That would indeed reset transit-visualization when navigating to this view via transit-changes. Problem is, you potentially can reah transit-visualization from other views and this fix does not cover that. As a general note transit-visualization related functionality should not leak into other components.

Reply:
+1
Generic design should be that a view initializes itself upon navigation and doesn't rely on other views - if it does, then it should enforce to prevent other navigation which might break the workflow. Which is not the case here :)